src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -767,8 +817,7 @@ int decryptFinal(ByteBuffer src, ByteBuffer dst)
         // scenario for the subsequent output buffer capacity check.
         checkDataLength(0, len);
```

Comment:
Perhaps this can be moved up to the beginning of this method just like the dst size check?

Reply:
You don't know what 'len', which includes ibuffer data, until this point in the code.

Reply:
Well, didn't you already calcuated the 'len' value in the beginning of the copied-from-GCTR RuntimeException check?

Reply:
I'm removing the runtime check in the next update, it was duplicate of the ShortBufferException.  Calculating 'len' through the if() above line 818 while it is prepares the bytebuffers is good enough.